cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/StreamFileJanitor.java

```diff
@@ -115,4 +120,12 @@ void clean(StreamConfig config, long currentTime) throws IOException {
   private boolean isPartitionDirector(Location location) throws IOException {
     return (location.isDirectory() && location.getName().indexOf('.') > 0);
   }
+
+  private boolean isStreamExists(Id.Stream streamId) throws IOException {
```

Comment:
`cleanAll` itself throws `IOException`. The caller needs to handle that. In production code, the caller is actually a executor thread that schedule to do cleanup and it just log the exception and schedule it again (hence if it's interim failure, it'll get retried when the janitor runs again in next scheduled time).